plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -56,81 +103,198 @@ protected Control createContents(Composite parent) {
     createPromoteSection(container);
 
     createAdvancedSection(container);
-    
-    Dialog.applyDialogFont(container); 
-    
+
+    Dialog.applyDialogFont(container);
+
+    loadPreferences();
+
+    setupDataBinding();
+
     return container;
   }
 
-  private void createProjectIdSection(Composite parent) {
-    Composite projectIdComp = new Composite(parent, SWT.NONE);
-    projectIdComp.setLayout(new GridLayout(3, false));
-    projectIdComp.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false));
-  
-    promptForProjectIdButton = new Button(projectIdComp, SWT.CHECK);
-    promptForProjectIdButton.setText(Messages.getString("deploy.prompt.projectid"));
-    promptForProjectIdButton.setLayoutData(new GridData(SWT.LEFT, SWT.CENTER, true, false, 3, 1));
-    promptForProjectIdButton.addSelectionListener(new SelectionListener() {
-      
+  private void setupDataBinding() {
+    bindingContext = new DataBindingContext();
+
+    setupProjectIdDataBinding(bindingContext);
+    setupProjectVersionDataBinding(bindingContext);
+    setupAutoPromoteDataBinding(bindingContext);
+    setupBucketDataBinding(bindingContext);
+
+    PreferencePageSupport.create(this, bindingContext);
+    bindingContext.updateTargets();
+    observables = new ObservablesManager();
+    observables.addObservablesFromContext(bindingContext, true, true);
+  }
+
+  private void setupProjectIdDataBinding(DataBindingContext dbc) {
+    final ISWTObservableValue overrideObservable =
```

Comment:
Can we do ISWTObservableValue<Boolean>?

The superinterface IObservable is generic. I'm not sure how that passes down to the child.